bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/source/SourceViewer.java

```diff
@@ -543,7 +546,12 @@ public void configure(SourceViewerConfiguration configuration) {
 		String[] types= configuration.getConfiguredContentTypes(this);
 		for (String t : types) {
 
-			doSetAutoEditStrategies(configuration.getAutoEditStrategies(this, t), t);
+			IAutoEditStrategy[] autoEditStrategies= configuration.getAutoEditStrategies(this, t);
+			List<IAutoEditStrategy> autoEditStrategiesList= new ArrayList<>(Arrays.asList(autoEditStrategies));
+			autoEditStrategiesList.add(new SurroundWithBracketsStrategy(this));
```

Comment:
Adding the strategy here makes it incredibly hard to disable. Why did you opt for this solution instead of returning it from `org.eclipse.jface.text.source.SourceViewerConfiguration.getAutoEditStrategies(ISourceViewer, String)` as a default?

Reply:
That is a good suggestion.
@lathapatil can you please consider the suggested change in a next PR ?